### DISCUSSION TITLE
Allow LDAP user to log in with email

### DISCUSSION THREAD

**Discussion by marcules** [directus/directus#12845] [Archive]
Hi,

currently I have the `uid` attribute configured as `_USER_ATTRIBUTE`. I can log in with the uid but not wit the email address of the user with the ldap provider - is that intended behaviour? Would it be feasible / possible to extend it so it accepts multiple identifiers (or an ldap-filter like `(|(uid=xxx)(mail=xxx))`?

Or am I missing some kind of configuration?

LG Marc

**Comment by aidenfoxx** [comment]
You just need to change the user attribute to whatever mail attribute you use for your user `AUTH_<PROVIDER>_USER_ATTRIBUTE="mail"`.

It is not possible to specify multiple attributes.

**Comment by marcules** [comment]
Is selecting multiple attributes something that you would like to support / have implemented? I could create a PR for that - if not, then this is also fine, I was just thinking, that our users usually "mix" which username they use (their "windows username" and/or their "email address") also this would probably allow for logging in with "alternative mails" configured in ldap if implemented as a configurable filter.

**Reply by aidenfoxx** [reply]
I don't think allowing custom filters is an option. Exposing the `ldapjs` functions we use for filtering to ensure correct formatting and escaping would be a nightmare, and allowing users to specify their own string filters in the config could be a security risk.

In that regard the simpler solution would be to allow the "_USER_ATTRIBUTE" variable to accept an array of attributes and generate the appropriate LDAP filter.

Since this is something that will only benefit a small minority of users it's not something I would be looking to implement. But a PR from your side is always welcome.
